cmd/promtool/main.go

```diff
@@ -283,8 +293,33 @@ func main() {
 	}
 }
 
+type lintSettings map[string]struct{}
```

Comment:
Instead of storing these at strings, I think it might be simpler to handle them like we do feature flags, which is a struct containing booleans for all the options.